### PR TITLE
Revert module directives.

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
## Description of the changes

To fix https://github.com/optuna/optuna/issues/1871

### CODE REVIEW

**Comment by keisuke-umezawa** [comment]
I also want to add `.. module:: optuna.pruners`. But if I added it, I got following errors:

```
/Users/k-umezawa/dev/optuna/docs/source/reference/pruners.rst.rst:6: WARNING: autosummary: stubファイルが見つかりません'optuna.pruners.optuna.pruners.BasePruner'。autosummary_gen
erate設定を確認してください。                                                                                                                                                     
/Users/k-umezawa/dev/optuna/docs/source/reference/pruners.rst.rst:6: WARNING: autosummary: stubファイルが見つかりません'optuna.pruners.optuna.pruners.MedianPruner'。autosummary_g
enerate設定を確認してください。                                                                                                                                                   
/Users/k-umezawa/dev/optuna/docs/source/reference/pruners.rst.rst:6: WARNING: autosummary: stubファイルが見つかりません'optuna.pruners.optuna.pruners.NopPruner'。autosummary_gene
rate設定を確認してください。                                                                                                                                                      
/Users/k-umezawa/dev/optuna/docs/source/reference/pruners.rst.rst:6: WARNING: autosummary: stubファイルが見つかりません'optuna.pruners.optuna.pruners.PercentilePruner'。autosumma
ry_generate設定を確認してください。                                                                                                                                               
/Users/k-umezawa/dev/optuna/docs/source/reference/pruners.rst.rst:6: WARNING: autosummary: stubファイルが見つかりません'optuna.pruners.optuna.pruners.SuccessiveHalvingPruner'。au
tosummary_generate設定を確認してください。                                                                                                                                        
/Users/k-umezawa/dev/optuna/docs/source/reference/pruners.rst.rst:6: WARNING: autosummary: stubファイルが見つかりません'optuna.pruners.optuna.pruners.HyperbandPruner'。autosummar
y_generate設定を確認してください。                                                                                                                                                
/Users/k-umezawa/dev/optuna/docs/source/reference/pruners.rst.rst:6: WARNING: autosummary: stubファイルが見つかりません'optuna.pruners.optuna.pruners.ThresholdPruner'。autosummar
y_generate設定を確認してください。                                                                                                                                                
更新されたファイルを探しています... 見つかりませんでした                                 
環境データを保存中... 完了                                                               
整合性をチェック中... /Users/k-umezawa/dev/optuna/docs/source/reference/generated/optuna.pruners.BasePruner.rst: WARNING: ドキュメントはどの toctree にも含まれていません         
/Users/k-umezawa/dev/optuna/docs/source/reference/generated/optuna.pruners.HyperbandPruner.rst: WARNING: ドキュメントはどの toctree にも含まれていません                          
/Users/k-umezawa/dev/optuna/docs/source/reference/generated/optuna.pruners.MedianPruner.rst: WARNING: ドキュメントはどの toctree にも含まれていません                             
/Users/k-umezawa/dev/optuna/docs/source/reference/generated/optuna.pruners.NopPruner.rst: WARNING: ドキュメントはどの toctree にも含まれていません                                
/Users/k-umezawa/dev/optuna/docs/source/reference/generated/optuna.pruners.PercentilePruner.rst: WARNING: ドキュメントはどの toctree にも含まれていません                         
/Users/k-umezawa/dev/optuna/docs/source/reference/generated/optuna.pruners.SuccessiveHalvingPruner.rst: WARNING: ドキュメントはどの toctree にも含まれていません                  
/Users/k-umezawa/dev/optuna/docs/source/reference/generated/optuna.pruners.ThresholdPruner.rst: WARNING: ドキュメントはどの toctree にも含まれていません 
```

🤔

**Review by crcrpar — COMMENTED** [review]
What do you think about reverting module to `multi_objective`?

Ref: https://github.com/optuna/optuna/commit/c7d5170d4d37720a8fa2ff1162528775a752de9c

File: docs/source/reference/integration.rst [hunk 1]
```diff
@@ -1,3 +1,5 @@
+.. module:: optuna.integration
+
 .. _integration_list:
```
Comment (crcrpar) [inline docs/source/reference/integration.rst:3]: Now `.. _integration_list` is not necessary now. I think it better to remove this and update https://github.com/optuna/optuna/blob/d809aacd0384a17d06dcab03cf99d5c4c86abc92/tutorial/007_pruning.py#L79.

**Comment by keisuke-umezawa** [comment]
If I deleted `import optuna`, the error disappeared. 🤔

**Review by HideakiImamura — COMMENTED** [review]
Thanks for the PR! The changes almost LGTM. Recently, `optuna.visualization.matplotlib` module is added. (https://github.com/optuna/optuna/pull/1869) Could you add `.. module:: optuna.visualization.matplotlib`?

[Just a note] To avoid errors for `pruners` and `multi_objective`, you delete `import optuna`. In the internal discussion with core-dev members, we suspect that that circular references in various parts of Optuna may be the cause. We will resolve these circular references as refactoring tasks in the future.

**Comment by keisuke-umezawa** [comment]
@HideakiImamura 
Sorry for being late. I followed your comment.

**Review by HideakiImamura — APPROVED** [review]
Thanks for the update! LGTM!

**Review by crcrpar — APPROVED** [review]

**Pull request merged**